cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -122,6 +133,24 @@ public static DataprocClient fromConf(DataprocConf conf) throws IOException, Gen
     String subnet = conf.getSubnet();
     Network networkInfo = getNetworkInfo(projectId, network, compute);
 
+    boolean useInternalIP;
+    
+    // if a peered network is specified then the instance is tied to a network
+    PeeringState state = peeringState(systemNetwork, systemProjectId, networkInfo);
+    if (state == PeeringState.ACTIVE) {
+      if (conf.isPreferExternalIP()) {
+        LOG.warn("Prefer external IP setting will be ignored and internal IP will used to communicate with the " +
+                   "Dataproc cluster as active peering with the system network found.");
+      }
+      useInternalIP = true;
+    } else if (state == PeeringState.INACTIVE) {
+      throw new IllegalStateException(String.format("VPC Peering to the system network %s in system project %s found " +
+                                                      "in INACTIVE state. Please fix peering setup to be in ACTIVE " +
+                                                      "state.", systemNetwork, systemProjectId));
+    } else {
```

Comment:
In case of private instance which is tied to a network and can only use private ip if the customer has not setup the peering on their end then the peeringState will return NONE in that case internalIP will be preferred unless someone set externalIP to true but the run will fail since there is no peering from customer side. 

I believe we will have to ensure that the peering does exist when instance is private.

Reply:
Please open a bug for this .